source/location_access.cpp

```diff
@@ -741,17 +822,24 @@ void AreaTable_Init() {
                   EventAccess(&BrokenSwordAccess, {[]{return IsAdult && (PoachersSawAccess || PoachersSaw);}}),
                 }, {
                   //Locations
-                  LocationAccess(GV_CHEST,          {[]{return CanUse(MEGATON_HAMMER) || (LogicGVHammerChest && IsAdult);}}),
+                  LocationAccess(GV_CHEST,          {[]{return IsAdult && (CanUse(MEGATON_HAMMER) || LogicGVHammerChest);},
+                                         /*Glitched*/[]{return IsAdult && (CanDoGlitch(GlitchType::LedgeCancel, GlitchDifficulty::NOVICE) || CanDoGlitch(GlitchType::BombHover, GlitchDifficulty::INTERMEDIATE) || (CanUse(STICKS) && CanDoGlitch(GlitchType::QPA, GlitchDifficulty::NOVICE) && (CanTakeDamageTwice || CanDoGlitch(GlitchType::QPA, GlitchDifficulty::ADVANCED))));}}),
                   LocationAccess(GV_TRADE_SAW,      {[]{return IsAdult && PoachersSaw;}}),
-                  LocationAccess(GV_GS_BEHIND_TENT, {[]{return CanUse(HOOKSHOT) && AtNight && CanGetNightTimeGS;}}),
-                  LocationAccess(GV_GS_PILLAR,      {[]{return CanUse(HOOKSHOT) && AtNight && CanGetNightTimeGS;}}),
+                  LocationAccess(GV_GS_BEHIND_TENT, {[]{return IsAdult && CanUse(HOOKSHOT) && AtNight && CanGetNightTimeGS;},
+                                         /*Glitched*/[]{return (CanDoGlitch(GlitchType::BombHover, GlitchDifficulty::INTERMEDIATE) || CanUse(BOOMERANG)) && IsAdult && AtNight && CanGetNightTimeGS;}}),
```

Comment:
Speaking of Gerudo Valley. Should `CanUse(BOOMERANG)` maybe go into the glitchless func above? (Same for any similar instances.)

Also regarding opening grottos with hammer, maybe that should be a Glitch Trick so it's not just blindly assumed, and is more clear to anyone willing to try this Glitched logic?

Reply:
Ah yeah, that's part of what I meant when I said 
> Not all situations are accounted for but nothing should cause an unbeatable seed
I do plan to go back through at some point and make sure all the item usability changes are considered everywhere but my initial pass was just making sure none of it inferred an age check. (That specific line is left over from when item usability was going to be only for glitch logic, but there are other similar cases)

Also I don't remember adding hammer to any grotto entrances that weren't already using CanBlastOrSmash, though I could be wrong.

Reply:
Is was thinking about how, for example, `KAKARIKO_VILLAGE` to `KAK_REDEAD_GROTTO` checks `CanUse(MEGATON_HAMMER)` in the Glitched function, but upon looking closer I see `CanUse(MEGATON_HAMMER)` is already checked in `CanBlastOrSmash`, so I suppose this counts as what you were going to look over again. So all good. 👍